src/app/shared/checker-workflow.service.ts

```diff
@@ -29,140 +29,169 @@ import { WorkflowService } from './workflow.service';
+        this.goToEntry(this.publicPage, workflow.full_workflow_path, 'workflow');
+      }, error => this.containersService.getPublishedContainer(parentId).pipe(first()).subscribe((tool: DockstoreTool) => {
+        this.goToEntry(this.publicPage, tool.tool_path, 'tool');
+      }, error2 => console.log('Can not get parent entry')));
+    } else {
+      this.workflowsService.getWorkflow(parentId).pipe(first()).subscribe((workflow: Workflow) => {
+        this.goToEntry(this.publicPage, workflow.full_workflow_path, 'workflow');
+      }, error => this.containersService.getContainer(parentId).pipe(first()).subscribe((tool: DockstoreTool) => {
+        this.goToEntry(this.publicPage, tool.tool_path, 'tool');
+      }, error2 => console.log('Can not get parent entry')));
     }
+  }
 
-    /**
-     * Go to entry
-     * @param path Full path of entry
-     * @param entryType Either 'tool' or 'workflow'
-     */
-    public goToEntry(publicPage: boolean, path: string, entryType: 'tool' | 'workflow'): void {
-        let url: string;
-        if (publicPage) {
-            url = '/' + entryType + 's/' + path;
-        } else {
-            url = '/my-' + entryType + 's/' + path;
-        }
-        this.router.navigateByUrl(url);
+  /**
+   * Go to entry
+   * @param path Full path of entry
+   * @param entryType Either 'tool' or 'workflow'
+   */
+  public goToEntry(publicPage: boolean, path: string, entryType: 'tool' | 'workflow'): void {
+    this.router.navigateByUrl(this.getEntryURL(publicPage, path, entryType));
+  }
+
+  /**
+   * Get the entry's URL
+   *
+   * @param {boolean} publicPage    Whether the user is currently on a public page or not
+   * @param {string} path           The entry path
+   * @param {('tool' | 'workflow')} entryType  Whether entry is a tool or workflow
+   * @returns {string}              The URL of the entry
+   * @memberof CheckerWorkflowService
+   */
+  public getEntryURL(publicPage: boolean, path: string, entryType: 'tool' | 'workflow'): string {
+    let url: string;
+    if (publicPage) {
+      url = '/' + entryType + 's/' + path;
+    } else {
+      url = '/my-' + entryType + 's/' + path;
     }
+    return url;
+  }
 
-    /**
-     * Determines if a given entry is a workflow or not (tool).
-     * Returns true if entry is a workflow, returns false if entry is not a workflow (i.e. a tool)
-     * @param entry The entry to check
-     */
-    public isEntryAWorkflow(entry: Entry): boolean {
-        if (!entry) {
-            return null;
-        }
-        return entry.hasOwnProperty('is_checker');
+  /**
+   * Determines if a given entry is a workflow or not (tool).
+   * Returns true if entry is a workflow, returns false if entry is not a workflow (i.e. a tool)
+   * @param entry The entry to check
+   */
+  public isEntryAWorkflow(entry: Entry): boolean {
+    if (!entry) {
+      return null;
     }
+    return entry.hasOwnProperty('is_checker');
```

Comment:
Might be more robust with an instanceof or similar 
https://basarat.gitbooks.io/typescript/docs/types/typeGuard.html

Reply:
was my initial attempt when i first made this function back then.  Had/having Compiler errors probably similar issue to

https://stackoverflow.com/a/46599798/4484759

Reply:
Is this service working with JSON rather than the objects out of swagger-codegen?

Reply:
This might be easier to understand: https://stackoverflow.com/questions/14425568/interface-type-check-with-typescript